### DISCUSSION TITLE
[ Audio ] Why Winamp?

### DISCUSSION THREAD

**Discussion by malvarenga123** [julianxhokaxhiu/FFNx#172] [Q&A]
Hello, I think it's great that you decided to use a plugin system for audio. It's definitely the most flexible way.
I just wanted to know why did you choose Winamp for it.
Sure it has many plugins, but most of them haven't been updated in quite a few years, so if there's a problem with the plugin, nobody will be able to properly fix it.
Maybe support for foobar2000 plugins could also be added? Or maybe you guys could get in touch with [kode54](https://github.com/kode54). He has been writing audio plugins for a lot of projects for many years, including porting vgmstream to foobar, so I think he could give you guys some more ideas about this.
I'm looking forward to the development of FFNx!

**Comment by julianxhokaxhiu** [comment]
I'm definitely open to consider this layer via PR!

At the moment we implemented WInamp because we had a need to use a closed source plugin named `in_psf.dll` that is used to reproduce the MINIPSF files.

Although I know there is a FOSS plugin version of the very same plugin for FooBar, This means that if someone is open to implement a FooBar layer on FFNx we can add it as a new layer :)

Thank you in advance!

**Comment by myst6re** [comment]
Couldn't agree more, I confirm that this Winamp layer is an easy way to make PSF works.

**Comment by malvarenga123** [comment]
Thank you for clarifying the reason!
It makes sense that you wanted to use a specific plugin and the layer was the most simple way to implement it, with the advantage of supporting other plugins also.
